crypto/src/signer.rs

```diff
@@ -0,0 +1,87 @@
+// Licensed under the Apache-2.0 license
+
+use crate::{AlgLen, CryptoError};
+
+/// An ECDSA signature
+pub struct EcdsaSig {
+    pub r: CryptoBlob,
+    pub s: CryptoBlob,
+}
+
+impl EcdsaSig {
+    pub fn default(alg: AlgLen) -> EcdsaSig {
+        EcdsaSig {
+            r: CryptoBlob::default(alg),
+            s: CryptoBlob::default(alg),
+        }
+    }
+}
+
+/// An ECDSA public key
+pub struct EcdsaPub {
+    pub x: CryptoBlob,
+    pub y: CryptoBlob,
+}
+
+impl EcdsaPub {
+    pub fn default(alg: AlgLen) -> EcdsaPub {
+        EcdsaPub {
+            x: CryptoBlob::default(alg),
+            y: CryptoBlob::default(alg),
+        }
+    }
+}
+
+/// An HMAC Signature
+pub type HmacSig = CryptoBlob;
+
+/// An HMAC Key
+pub type HmacKey = CryptoBlob;
+
+/// A common base struct that can be used for all digests, signatures, and keys.
+pub struct CryptoBlob {
```

Comment:
minor nit: I think I prefer CryptoBuf. Blob implies to me that the caller shouldn't interpret the data

Reply:
I like that name better too.

Done.